hardware/arduino/avr/cores/arduino/Tone.cpp

```diff
@@ -456,19 +457,19 @@ void disableTimer(uint8_t _timer)
 
 #if defined(TIMSK3)
     case 3:
-      TIMSK3 = 0;
+      TIMSK3 &= ~(1 << OCIE3A);
```

Comment:
Did you test it with a Leonardo? 
If so, it would be more coherent to use the notation `bitWrite(TIMSK3, OCIE3A, 0);` as in the previous case